### PR TITLE
operations: add CleanDeleteVirtualMachine

### PR DESCRIPTION
**Pull request by colemickens**
- add new ACSEngineClient functions as needed
- add new interface for the AzureStorageClient
  (it's a separate interface because there might be more
  than one storage account used in the context of a request)

`CleanDeleteVirtualMachine` deletes a VM and the associated NIC and OS Disk blob.

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/634)

### CODE REVIEW

File: pkg/armhelpers/storage.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+// DeleteBlob deletes the specified blob
+// TODO(colemick): why doesn't SDK give a way to just delete a blob by URI?
+// it's what it ends up doing internally anyway...
+func (as *AzureStorageClient) DeleteBlob(vhdContainer, vhdBlob string) error {
+	bs := as.client.GetBlobService()
+	containerRef := bs.GetContainerReference(vhdContainer)
+	blobRef := containerRef.GetBlobReference(vhdBlob)
+
```
Comment (JackQuincy) [inline pkg/armhelpers/storage.go]: Does the skills specify page or block? or is that handled internally?

Comment (JackQuincy) [inline pkg/armhelpers/storage.go]: SDK not skills... Autocorrect 😀

Comment (colemickens) [inline pkg/armhelpers/storage.go]: Doesn't matter. When the call hits the wire it's just a generic HTTP DELETE against the blob URI. Same semantics for block/page blobs.

**Comment by colemickens** [comment]
@JackQuincy Can you elaborate on this: *"Though I'd like a unit test validating the passing around of the os disk uri in the vm delete method."* I do have a unit test that checks the proper decomposition of the VHD URI... are you looking for something beyond that?

**Comment by colemickens** [comment]
@acs-bot test this please

**Comment by JackQuincy** [comment]
Yes and that was great I just wanted in level higher as well. Where you mock the responses from the clients and pass things between the clients. Maybe that is more important when we start doing more error handling. Understood if it is it out of scope.

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________

**Comment by colemickens** [comment]
@acs-bot test this please

**Comment by amanohar** [comment]
https://github.com/Azure/acs-engine/pull/634

**Pull request merged**